lib/Zonemaster/CLI.pm

```diff
@@ -626,39 +458,39 @@ sub run {
                     }
                 }
             }
-            if ( $self->stop_level and $numeric{ uc $entry->level } >= $numeric{ $self->stop_level } ) {
+            if ( $opt_stop_level and $numeric{ uc $entry->level } >= $numeric{$opt_stop_level} ) {
                 die( Zonemaster::Engine::Exception::NormalExit->new( { message => "Saw message at level " . $entry->level } ) );
             }
         }
     );
 
-    if ( scalar @{ $self->extra_argv } > 1 ) {
-        say STDERR __( "Only one domain can be given for testing. Did you forget to prepend an option with '--<OPTION>'?" );
+    if ( @argv > 1 ) {
+        say STDERR __(
+            "Only one domain can be given for testing. Did you forget to prepend an option with '--<OPTION>'?" );
         return $EXIT_USAGE_ERROR;
     }
-
-    my ( $domain ) = @{ $self->extra_argv };
-
-    if ( !defined $domain ) {
+    elsif ( @argv < 1 ) {
         say STDERR __( "Must give the name of a domain to test." );
         return $EXIT_USAGE_ERROR;
     }
 
+    my ( $domain ) = @argv;
+
     ( my $errors, $domain ) = normalize_name( decode( 'utf8', $domain ) );
 
-    if ( $self->ns and @{ $self->ns } > 0 ) {
+    if ( @opt_ns ) {
         local $@;
         eval {
-            $self->check_fake_delegation( $domain );
+            check_fake_delegation( $domain, @opt_ns );
             1;
         } or do {
             print STDERR $@;
             return $EXIT_USAGE_ERROR;
         };
     }
 
-    if ( $self->ds and @{ $self->ds } ) {
-        $self->check_fake_ds( $domain );
+    if ( @opt_ds ) {
+        check_fake_ds( $domain, @opt_ds );
```

Comment:
I immediately found a bug here that wasn't caught by the unit tests. I'll see if I can add a unit test to cover it or otherwise I'll add an item to the testing section of the PR description.